core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -268,20 +325,24 @@ public Snapshot currentSnapshot() {
     return snapshotLog;
   }
 
+  public List<MetadataLogEntry> previousMetadata() {
```

Comment:
Just thinking out loud: is this name the most descriptive?

Is it just me who would expect `previousMetadata` in `TableMetadata` to return another instance of `TableMetadata`? Isn't the current logic more about previous metadata files rather than metadata in general? This applies to all places, like `addPreviousMetadata`, `deleteRemovedMetadata`, etc.

Reply:
@aokolnychyi Not sure if I got it correctly.

The `previousMetadata` is very similar to `snapshotLog`, I think returning List<MetadataLogEntry> make sense to me. I'm fine with any name.

The initial implementation of `addPreviousMetadata` was to return another instance of `TableMetadata`, but moving the logic of handling previous versions to `TableMetadata` is better I believe.

The `deleteRemovedMetadata` is not changing the state of `TableMetadata`, it's better not to return another instance of `TableMetadata`.

Reply:
No-no, I was referring to the name only. My point is that the current name is a bit confusing to me.

Reply:
For example, `previousMetadata` doesn't really return a previous metadata instance. Instead, it actually returns a list of previous metadata files. So, even `previousMetadataFiles` sounds a bit better to me.

Reply:
It might be only me, so leave it up to you.

Reply:
I noticed this as well, but this is not a public API so we can change it later.